src/components/biggive-campaign-card-filter-grid/biggive-campaign-card-filter-grid.tsx

```diff
@@ -196,7 +196,8 @@ export class BiggiveCampaignCardFilterGrid {
     for (const filterKey of Object.keys(filters)) {
       // https://stackoverflow.com/a/69757191/2803757
       const filterValue: string = filters[filterKey as keyof typeof filters];
-      if (filterValue?.length === 0) {
+
+      if (filterValue === null || filterValue.length === 0) {
```

Comment:
I found by adding console.log that filterValue was being null. But now I'm not sure why typescript is allowing it to be null when it's typed as `string`. We have `strictNullChecks` turned on already in tsconfig

Reply:
@bdsl I think it's the ts-ignore here that's letting the nulls sneak in. https://github.com/thebiggive/components/blob/08f1f88280213c8c97646db2ce3378c208b2e821/src/components/biggive-campaign-card-filter-grid/biggive-campaign-card-filter-grid.tsx#L169

Reply:
@bdsl I don't think our use is illegal based on that rule because the choice of operator guaranteed we were factoring in the possibility of a `null`.

However, the way I did so made no sense. I think the `.length` of a null var's nullable-dot expression is length of undefined, which is also undefined – not 0.

Reply:
@NoelLH Yep you're right - I guess I would have liked the compiler to complain the other way then - to say that the null safe operator is superfluous since `filterValue` is typed as string. It's the ts-ignore that allowed it to be typed as string when at runtime it was nullable.